game-app/game-core/src/main/java/games/strategy/triplea/delegate/EditDelegate.java

```diff
@@ -172,21 +172,35 @@ public String changeTerritoryOwner(final Territory territory, final GamePlayer p
 
   @Override
   public String changePUs(final GamePlayer player, final int newTotal) {
+    return changeResource(player, Constants.PUS, newTotal);
+  }
+
+  @Override
+  public String changeResource(
+      final GamePlayer player, final String resourceName, final int newTotal) {
     final String result = checkEditMode();
     if (result != null) {
       return result;
     }
-    final Resource pus = getData().getResourceList().getResource(Constants.PUS);
-    final int oldTotal = player.getResources().getQuantity(pus);
+    final Resource resource = getData().getResourceList().getResource(resourceName);
+    final int oldTotal = player.getResources().getQuantity(resource);
     if (oldTotal == newTotal) {
-      return "New PUs total is unchanged";
+      return "New " + resourceName + " total is unchanged";
     }
     if (newTotal < 0) {
```

Comment:
Would it make sense to zero this out, or change the error message to say the new total must be zero or greater (rather than simply "invalid")

Reply:
Done. Changed to "must not be negative".